qt/scientific_interfaces/Indirect/test/IndirectSpectrumSelectionPresenterTest.h

```diff
@@ -0,0 +1,367 @@
+  }
+
+  /// Public methods
+  MOCK_CONST_METHOD0(minimumSpectrum, std::size_t());
+  MOCK_CONST_METHOD0(maximumSpectrum, std::size_t());
+
+  MOCK_CONST_METHOD0(spectraString, std::string());
+  MOCK_CONST_METHOD0(maskString, std::string());
+
+  MOCK_METHOD1(displaySpectra, void(std::string const &spectraString));
+  MOCK_METHOD2(displaySpectra, void(int minimum, int maximum));
+
+  MOCK_METHOD2(setSpectraRange, void(int minimum, int maximum));
+
+  MOCK_METHOD0(showSpectraErrorLabel, void());
+  MOCK_METHOD0(hideSpectraErrorLabel, void());
+
+  MOCK_METHOD1(setMaskSelectionEnabled, void(bool enabled));
+  MOCK_METHOD0(clear, void());
+
+  /// Public slots
+  MOCK_METHOD1(setMinimumSpectrum, void(std::size_t spectrum));
+  MOCK_METHOD1(setMaximumSpectrum, void(std::size_t spectrum));
+
+  MOCK_METHOD1(setSpectraString, void(std::string const &spectraString));
+  MOCK_METHOD1(setMaskString, void(std::string const &maskString));
+};
+
+/// Note that there is limited (if any) interaction going from this model to the
+/// IndirectSpectrumSelectionView, meaning that not many methods are required
+/// for mocking.
+class MockIndirectFittingModel : public IndirectFittingModel {
+public:
+  /// Public methods
+  MOCK_CONST_METHOD2(getExcludeRegion,
+                     std::string(std::size_t dataIndex, std::size_t index));
+  MOCK_CONST_METHOD0(isMultiFit, bool());
+
+private:
+  std::string sequentialFitOutputName() const override { return ""; };
+  std::string simultaneousFitOutputName() const override { return ""; };
+  std::string singleFitOutputName(std::size_t index,
+                                  std::size_t spectrum) const override {
+    UNUSED_ARG(index);
+    UNUSED_ARG(spectrum);
+    return "";
+  };
+
+  std::vector<std::string> getSpectrumDependentAttributes() const override {
+    return {};
+  };
+};
+
+GNU_DIAG_ON_SUGGEST_OVERRIDE
+
+class IndirectSpectrumSelectionPresenterTest : public CxxTest::TestSuite {
+public:
+  /// Needed to make sure everything is initialized
+  IndirectSpectrumSelectionPresenterTest() { FrameworkManager::Instance(); }
+
+  static IndirectSpectrumSelectionPresenterTest *createSuite() {
+    return new IndirectSpectrumSelectionPresenterTest();
+  }
+
+  static void destroySuite(IndirectSpectrumSelectionPresenterTest *suite) {
+    delete suite;
+  }
+
+  void setUp() override {
+    m_view = new NiceMock<MockIndirectSpectrumSelectionView>();
+    m_model = new NiceMock<MockIndirectFittingModel>();
+    m_presenter = new IndirectSpectrumSelectionPresenter(m_model, m_view);
+
+    SetUpADSWithWorkspace ads("WorkspaceName", createWorkspace(10));
+    m_model->addWorkspace("WorkspaceName");
+  }
+
+  void tearDown() override {
+    AnalysisDataService::Instance().clear();
+
+    TS_ASSERT(Mock::VerifyAndClearExpectations(m_view));
+    TS_ASSERT(Mock::VerifyAndClearExpectations(m_model));
+
+    delete m_presenter;
+    delete m_model;
+    // delete m_view; - causes an error
```

Comment:
I'm OK leaving this out for now given that having these interfaces tested is definitely needed, however can you add an issue to fix this and reference the issue number in the comment (or add some more details to the comment if you have any insight into what is causing the error).